app/assets/src/components/views/bulk_download/Chooser.jsx

```diff
@@ -183,11 +172,85 @@ class ChooseStep extends React.Component {
     orderBy(["sampleCount", "text"], ["desc", "asc"], options)
   );
 
+  // *** BULK DOWNLOAD GENERATION ***
```

Comment:
What does this comment indicate? a lot of the functions below are just handlers, right?